pkg/netdevice/pciNetDevice.go

```diff
@@ -95,24 +100,47 @@ func NewPciNetDevice(dev *ghw.PCIDevice,
 		return nil, err
 	}
 
+	var ddpFunc ddpProfileGetFunc
+
+	if utils.IsDevlinkDDPSupportedByPCIDevice(netDev.GetPfPciAddr()) {
+		ddpFunc = utils.DevlinkGetDDPProfiles
+	} else if utils.IsDDPToolSupportedByDevice(netDev.GetPfPciAddr()) {
+		ddpFunc = utils.GetDDPProfiles
+	} else {
+		ddpFunc = unsupportedDDP
+	}
+
 	return &pciNetDevice{
-		HostDevice:   hostDev,
-		GenPciDevice: *pciDev,
-		GenNetDevice: *netDev,
-		vdpaDev:      vdpaDev,
+		HostDevice:    hostDev,
+		GenPciDevice:  *pciDev,
+		GenNetDevice:  *netDev,
+		vdpaDev:       vdpaDev,
+		getDDPProfile: ddpFunc,
```

Comment:
can we roll these changes into GetDDPProfiles ?  thus there would be no changes when creating PciNetDevice ?

Reply:
that way you also dont need to call ddpFuc twice in case of error at L124/127

Reply:
Yes we could move the check if devlink is supported to GetDDPProfiles. 
I will call it devlink query for clarity here:
we would still need:

call devlink query on pci address
call devlink query on the nd parent PF if the first attempt failed
call ddptool if both of the above fail

or am I missing something?

Reply:
sure so,

devlink query only works on PF pci address

so if it first fails on pci address, this could means it's a vf pci
so we get the parent pci address, and try again.
vf will share the pf ddp profile.

as for the ddptool, it works on pf and vf.
thus, only requiring 1 call.